File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/enumeration/Article.java
package org.babyfish.jimmer.sql.model.enumeration;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@DatabaseValidationIgnore
@Entity
public interface Article {

    @Id
    @GeneratedValue
    long id();

    String name();

    @ManyToOne
    Writer writer();

    @ManyToOne
    Approver approver();
}

File: project/jimmer-sql-test/jimmer-sql-test-model-kotlin/src/main/kotlin/org/babyfish/jimmer/sql/kt/model/enumeration/Article.kt
package org.babyfish.jimmer.sql.kt.model.enumeration

import org.babyfish.jimmer.sql.DatabaseValidationIgnore
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.ManyToOne

@DatabaseValidationIgnore
@Entity
interface Article {

    @Id
    val id: Long

    val name: String

    @ManyToOne
    val writer: Writer

    @ManyToOne
    val approver: Approver
}